server/src/main/java/org/elasticsearch/transport/TransportSettings.java

```diff
@@ -51,6 +52,11 @@
         key -> intSetting(key, -1, -1, Setting.Property.NodeScope));
     public static final Setting<Boolean> TRANSPORT_COMPRESS =
         boolSetting("transport.compress", false, Setting.Property.NodeScope);
+    public static final Setting<Boolean> TRANSPORT_COMPRESS_RAW_DATA =
+        boolSetting("transport.compress_raw_data", false, Setting.Property.NodeScope);
```

Comment:
Can we add validation that `compress_raw_data=true` can only be set if `compress=false`?